File: Chroma-Core/src/main/java/buttondevteam/lib/chat/Command2MCSender.java
package buttondevteam.lib.chat;

import buttondevteam.core.component.channel.Channel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.bukkit.command.CommandSender;

@RequiredArgsConstructor
public class Command2MCSender implements Command2Sender {
	private @Getter final CommandSender sender;
	private @Getter final Channel channel;
	private @Getter final CommandSender permCheck;

	@Override
	public void sendMessage(String message) {
		sender.sendMessage(message);
	}

	@Override
	public void sendMessage(String[] message) {
		sender.sendMessage(message);
	}

	@Override
	public String getName() {
		return sender.getName();
	}
}

File: Chroma-Core/src/main/java/buttondevteam/lib/chat/Command2MCSender.kt
package buttondevteam.lib.chat

import buttondevteam.core.component.channel.Channel
import buttondevteam.lib.player.ChromaGamerBase

class Command2MCSender(val sender: ChromaGamerBase, val channel: Channel, val permCheck: ChromaGamerBase) : Command2Sender {
    // TODO: Remove this class and only use the user classes.
    // TODO: The command context should be stored separately.
    override fun sendMessage(message: String) {
        sender.sendMessage(message)
    }

    override fun sendMessage(message: Array<String>) {
        sender.sendMessage(message)
    }

    override fun getName(): String {
        return sender.name
    }
}